sympy/solvers/polysys.py

```diff
@@ -429,3 +439,389 @@ def solve_triangulated(polys, *gens, **args):
 
         solutions = _solutions
     return sorted((s for s, _ in solutions), key=default_sort_key)
+
+
+def factor_poly_system(eqs, *gens, **kwargs):
+    """
+    Given a system of polynomial equations, returns a factored form and solvability conditions.
+
+    Parameters
+    ==========
+
+    eqs : list
+        List of expressions to be factored. Each expression is assumed to be equal to zero.
+    *gens : Symbol or sequence of Symbols, optional
+        Generator(s) of the polynomial ring. If not provided, all free symbols will be used.
+    **kwargs : dict, optional
+        Optional keyword arguments passed to polynomial manipulation functions.
+
+    Returns
+    =======
+
+    tuple
+        A pair (systems, condition) where:
+        - systems is a list of lists of expressions, where each sublist when solved gives
+          one component of the solution
+        - condition is a Boolean expression that must be satisfied for the system to be solvable
+
+    Examples
+    ========
+
+    >>> from sympy.solvers.polysys import factor_poly_system
+    >>> from sympy.abc import x, y
+
+    A simple system with multiple solutions:
+    >>> systems, cond = factor_poly_system([x**2 - 1, y - 1])
+    >>> systems
+    [[x + 1, y - 1], [x - 1, y - 1]]
+    >>> cond
+    True
+
+    A system with no solution:
+    >>> systems, cond = factor_poly_system([x, 1])
+    >>> systems
+    []
+    >>> cond
+    True
+
+    A system with infinite solutions:
+    >>> systems, cond = factor_poly_system([x - x, (x + 1)**2 - (x**2 + 2*x + 1)])
+    >>> systems
+    [[]]
+    >>> cond
+    True
+
+    Notes
+    =====
+
+    - A return of a list constain an empty list [[]] in the tuple means any value of the symbol(s) is a solution
+    - A return of an empty list [] in the tuple means no solutions for the system exists.
```

Comment:
There should be an Explanation section that explains that the solution set for the original system is the union of the solution sets of the factorized systems. These notes could be moved to that section.